### PR TITLE
Add PlatformIO support

### LINKED ISSUE #85: Add PlatformIO support
## Desired behavior
Add [PlatformIO](https://platformio.org/) support to be able to build and upload the firmware using Visual Studio Code (by using the PlatformIO extension). Among other things, some advantages from doing so would be:
- No further need to install the Arduino IDE.
- Allows to add unit testing.
- Allows to support development boards other than Arduino.

Arduino IDE support for building and uploading the code shall remain as an easy and straightforward option to do so.

### PR DESCRIPTION
**Pull request by jballoffet**
# 🎉 New feature

Closes #85

## Summary
This PR adds PlatformIO support by:
- Adding `platformio.ini` file.
- Modifying file structure to comply with PIO requirements (source files moved within a `src` directory).
- Renaming all .ino files to .cpp.
- Adding required `include` directives.
- Fixing warnings and errors thrown after the files renaming.

## Test it
This PR can be tested by opening the andino_firmware project either on Arduino IDE or PlatformIO (within VS Code) and then building and uploading the code.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: andino_firmware/andino_firmware.ino [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/*
+ * Arduino IDE sketch file. This file is required to be able to compile and
+ * upload the Andino firmware using Arduino IDE.
+ */
```
Comment (francocipollone) [inline andino_firmware/andino_firmware.ino:4]: I tried to verifying it using the Arduino IDE but given that this file doesn't have any reference to setup or loop it fails. 
Are you missing here a include?

Comment (francocipollone) [reply andino_firmware/andino_firmware.ino:4]: Shouldn't we do something like the following: https://community.platformio.org/t/make-platformio-project-compatible-with-arduino-ide/3368/10

Comment (jballoffet) [reply andino_firmware/andino_firmware.ino:4]: That's weird, it works just fine on my end. When building, Arduino IDE is able to compile and link all .ino and .cpp files within the sketch folder, up to one level of folder nesting (see Arduino preprocessing process here: https://arduino.github.io/arduino-cli/0.33/sketch-build-process/#pre-processing).

I saw the link you provided, but in order to make it Arduino-agnostic, I'd rather not rename main.cpp to main.ino. I have followed this thorough tutorial instead: https://community.platformio.org/t/tutorial-for-creating-multi-cpp-file-arduino-project/5830/38.

I have tested the building process with both Arduino IDE 2.1.0 and 1.8.19. What version are you using?

The only required steps to build are:
- Open Arduino IDE.
- Go to File > Open.
- Select andino_firmware.ino.
- Click on verify.

Let me know how that goes, thanks!

**Review by francocipollone — APPROVED** [review]
LGTM!! I really like how it is being organized now!

We need to clarify in the docs that the Arduino IDE to use is the +2.0 and not the legacy one.
In my case, I had an issue with that IDE and the Arduino Nano I am using. It got fixed by using 
image
That is something to add to the docs in case people find that issue.

**Comment by jballoffet** [comment]
Great, thanks @francocipollone! I will rebase and merge then. 🚀

**Review by francocipollone — APPROVED** [review]
:rocket:

**Pull request merged**